crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,705 @@
+use std::sync::RwLockWriteGuard;
+
+use crate::prelude::*;
+
+#[uniffi::export]
+impl SargonOS {
+    pub async fn analyse_transaction_preview(
+        &self,
+        instructions: String,
+        blobs: Blobs,
+        message: Message,
+        is_wallet_transaction: bool,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionToReview> {
+        self.perform_transaction_preview_analysis(
+            instructions,
+            blobs,
+            message,
+            is_wallet_transaction,
+            nonce,
+            notary_public_key,
+        )
+        .await
+    }
+}
+
+impl SargonOS {
+    pub async fn perform_transaction_preview_analysis(
+        &self,
+        instructions: String,
+        blobs: Blobs,
+        message: Message,
+        is_wallet_transaction: bool,
```

Comment:
I think we should use the term `host` instead of `wallet`. 

so maybe change `is_wallet_transaction` -> `are_instructions_originating_from_host` or `instructions_originates_from_host` or similar?